tools/update.php

```diff
@@ -0,0 +1,746 @@
+ *
+ * @param string $loris_root   The directory where LORIS is installed.
+ * @param string $version_from Pre-update LORIS version.
+ * @param string $version_to   Post-update LORIS version.
+ *
+ * @return array Patch files to be applied. Empty if up-to-date.
+ */
+function getPatchesFromVersion(
+    $loris_root,
+    $version_from,
+    $version_to
+): array {
+
+    // Definie indices corresponding to our semantic versioning system.
+    define('MAJOR', 0);
+    define('MINOR', 1);
+    define('BUGFIX', 2);
+    // Convert string representation of version into arrays of ints
+    $from_versions = array_map('intval', explode('.', $version_from));
+    $to_versions   = array_map('intval', explode('.', $version_to));
+    // Calculate difference between old version and latest
+    $diff_major = $to_versions[MAJOR] - $from_versions[MAJOR];
+    $diff_minor = $to_versions[MINOR] - $from_versions[MINOR];
+    /* Display output based on version differences. Actual patching occurs
+     * below.
+     */
+    if ($diff_major < 0) {
+        echo '[!] Your version of LORIS is ahead of the latest release. If you '
+            . 'are not a developer working on the bleeding edge of LORIS, then '
+            . 'something has gone very wrong. SQL patches will not be applied.'
+            . PHP_EOL;
+        return array();
+    } else if ($diff_major === 0) {
+        // If major version is equal, check for difference in minor releases.
+        if ($diff_minor < 0) {
+            /* Major versions equal, but installed minor version is ahead of the
+             * latest release.
+             */
+            echo '[!] Your version of LORIS is ahead of the latest release. If '
+                . 'you are not a developer working on the bleeding edge of '
+                . 'LORIS, then something has gone very wrong. SQL patches will '
+                . 'not be applied.'
+                . PHP_EOL;
+            return array();
+        }
+        if ($diff_minor > 0) {
+            echo "[**] Latest version $version_to is ahead of installed "
+                . "$version_from by $diff_minor MINOR release(s)."
+                . PHP_EOL;
+        } else {
+            // If major and minor differences are 0, then LORIS is up-to-date.
+            echo "[**] Major and minor versions are equal. No patches to "
+                . "apply."
+                . PHP_EOL;
+            return array();
+        }
+    } else {
+        echo "[**] Latest version $version_to is ahead of installed "
+            . "$version_from by $diff_major MAJOR release(s)"
+            . PHP_EOL;
+    }
+
+    /* Following a discussion on this pull request, it was decided that patching
+     * SQL files automatically when a LORIS is more than one version behind
+     * should not be supported. This is to minimize errors that could impact
+     * data integrity.
+     * https://github.com/aces/Loris/pull/3677#discussion_r202814374
+     */
+    if ($diff_major > 1) {
```

Comment:
now that I think of it it could also apply to multiple minor versions ? not sure

Reply:
It's still an extra command-line flag and the script prints out the release notes. Should we add more protections here?

Reply:
maybe not, i might need to give it a run to understand fully. are all the possible flags defined anywhere ? any examples on how to run the script/ a man page

Reply:
I added an explicit warning and question-prompt so that a user will be sure of what they're doing.

Reply:
It has a usage message when you use the script wrong, but no there are no examples. Should I add some? Feels like that's the sort of thing that should maybe go into a README file